File: step17example/src/main/java/com/example/step17example/TodoAdapter.java
package com.example.step17example;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.List;

public class TodoAdapter extends BaseAdapter {
    private Context context;
    private int layoutRes;
    private List<TodoDto> list;
    //생성자
    public TodoAdapter(Context context, int layoutRes, List<TodoDto> list){
        this.context=context; // 액티비티의 참조값
        this.layoutRes=layoutRes; // ListView 의 cell layout 리소스 아이디
        this.list=list; // ListView 에 출력할 데이터를 가지고 있는 모델
    }
    //모델의 갯수 리턴

    @Override
    public int getCount() {
        return list.size();
    }
    //position 에 해당하는 모델을 리턴
    @Override
    public Object getItem(int position) {
        return list.get(position);
    }
    //position 에 해당하는 아이템의 아이디(primary key)
    @Override
    public long getItemId(int position) {
        //position 에 해당하는 TodoDto 의 번호를 리턴해준다
        return list.get(position).getNum();
    }
    //position 에 해당하는 cell View 를 리턴해 주어야 한다. (ListView 가 호출할 예정인 메소드)
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        //만일 View 를 새로 만들어야 한다면
        if(convertView == null){
            //여기서 만들고 ( res/layout/listview_cell.xml 문서를 전개해서)
            //해당 문서의 리소스 아이디는 생성자의 인자로 전달되어서 필드에 저장해 놓은 상태
            convertView = LayoutInflater.from(context).inflate(layoutRes, parent, false);
        }
        // View 에 모델이 가지고 있는 데이터를 출력하고
        TodoDto dto=list.get(position);
        //listviiew_cell.xml 안에 있는 TextView 의 참조값 얻어와서 문자열 출력하기
        TextView text_content=convertView.findViewById(R.id.text_content);
        TextView text_regdate=convertView.findViewById(R.id.text_regdate);
        text_content.setText(dto.getContent());
        text_regdate.setText(dto.getRegdate());
        // View 를 리턴해준다.
        return convertView;
    }
}

File: step14sqlite/src/main/java/com/example/step14sqlite/TodoAdapter.kt
package com.example.step14sqlite

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

/*
    Context type, Int type, List<Todo> type 을 인자로 전달받는 대표 생성자
    인자로 전달받은 내용을 자동으로 필드에 저장된다.
    BaseAdapter 추상 클래스를 상속 받아서 만든다.
 */
class TodoAdapter(var context: Context, var layoutRes:Int, var list:List<Todo>) : BaseAdapter(){

    //전체 sell 의 갯수 리턴
    override fun getCount(): Int {
        return list.size
    }

    //인자로 전달된 position 에 해당하는 아이템 리턴하기
    override fun getItem(position: Int): Any {
        return list.get(position)
    }
    //인자로 전달된 position 에 해당하는 아이템의 아이디 리턴하기
    override fun getItemId(position: Int): Long {
        return list.get(position).num.toLong() //Int type 을 Long type 으로 casting 해서 리턴
    }
    //인자로 전달된 cell view 를 리턴하는 메소드
    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {

        //kotlin 에서 메소드의 매개변수는 상수 (val) 이기 때문에 값 변경이 불가하다
        var resultView:View = if(convertView == null)
            LayoutInflater.from(context).inflate(layoutRes, parent, false)
        else  convertView

        //position 에 해당하는 할일 정보를 얻어와서
        val data=list.get(position)

        //TextView 에 출력하고
        val text_content=resultView.findViewById<TextView>(R.id.text_content)
        val text_regdate=resultView.findViewById<TextView>(R.id.text_regdate)

        text_content.text=data.content
        text_regdate.text=data.regdate

        //해당 View 를 리턴해준다.
        return resultView
    }

}